CRM/Core/Form.php

```diff
@@ -2069,19 +2069,12 @@ public function addField($name, $props = [], $required = FALSE, $legacyDate = TR
    * @param string $usedFor
    */
   public function addProfileSelector($name, $label, $allowCoreTypes, $allowSubTypes, $entities, $default = FALSE, $usedFor = NULL) {
-    // Output widget
-    // FIXME: Instead of adhoc serialization, use a single json_encode()
-    CRM_UF_Page_ProfileEditor::registerProfileScripts();
-    CRM_UF_Page_ProfileEditor::registerSchemas(CRM_Utils_Array::collect('entity_type', $entities));
-    $this->add('text', $name, $label, [
-      'class' => 'crm-profile-selector',
-      // Note: client treats ';;' as equivalent to \0, and ';;' works better in HTML
-      'data-group-type' => CRM_Core_BAO_UFGroup::encodeGroupType($allowCoreTypes, $allowSubTypes, ';;'),
-      'data-entities' => json_encode($entities),
-      //CRM-15427
-      'data-default' => $default,
-      'data-usedfor' => json_encode($usedFor),
-    ]);
+    // @todo Filter by "used for"
+    $profileGroups = \Civi\Api4\UFGroup::get(FALSE)
```

Comment:
Probably we *should* check permissions here.

Reply:
Fixed (and I still need to fix the "used for" filtering)